File: core-java/src/main/java/com/kakao/actionbase/core/java/metadata/v3/common/DirectionType.java
package com.kakao.actionbase.core.java.metadata.v3.common;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public enum DirectionType {
  BOTH(Direction.OUT, Direction.IN),
  OUT(Direction.OUT),
  IN(Direction.IN),
  ;

  private final List<Direction> directions;

  DirectionType(Direction... dirs) {
    this.directions = new ArrayList<>(dirs.length);
    this.directions.addAll(Arrays.asList(dirs));
  }

  public static DirectionType of(String name) {
    return DirectionType.valueOf(name.toUpperCase());
  }

  public List<Direction> directions() {
    return directions;
  }
}

File: core/src/main/kotlin/com/kakao/actionbase/core/metadata/common/DirectionType.kt
package com.kakao.actionbase.core.metadata.common

enum class DirectionType(
    vararg dirs: Direction,
) {
    BOTH(Direction.OUT, Direction.IN),
    OUT(Direction.OUT),
    IN(Direction.IN),
    ;

    private val directions: List<Direction> = dirs.toList()

    fun directions(): List<Direction> = directions

    companion object {
        fun of(name: String): DirectionType = valueOf(name.uppercase())
    }
}
